src/coreclr/src/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -1281,15 +1281,30 @@ private static void ValidateModulePointer(RuntimeModule module)
         public RuntimeTypeHandle GetRuntimeTypeHandleFromMetadataToken(int typeToken) { return ResolveTypeHandle(typeToken); }
         public RuntimeTypeHandle ResolveTypeHandle(int typeToken)
         {
-            return new RuntimeTypeHandle(ResolveTypeHandleInternal(GetRuntimeModule(), typeToken, null, null));
+            return new RuntimeTypeHandle(ResolveTypeHandleInternal(GetRuntimeModule(), typeToken));
         }
         public RuntimeTypeHandle ResolveTypeHandle(int typeToken, RuntimeTypeHandle[]? typeInstantiationContext, RuntimeTypeHandle[]? methodInstantiationContext)
         {
             return new RuntimeTypeHandle(ModuleHandle.ResolveTypeHandleInternal(GetRuntimeModule(), typeToken, typeInstantiationContext, methodInstantiationContext));
         }
 
+        internal static RuntimeType ResolveTypeHandleInternal(RuntimeModule module, int typeToken)
+        {
+            ValidateModulePointer(module);
+            if (!ModuleHandle.GetMetadataImport(module).IsValidToken(typeToken))
```

Comment:
The up-front token validation is unnecessary. `ResolveType` QCall is going to validate the token and throw exception for bad tokens as well. It should be possible to avoid the cost by doing something like this:
```
try
{
    ResolveType(...);
}
catch (Exception e)
{
   if (!ModuleHandle.GetMetadataImport(module).IsValidToken(typeToken))
       throw new ArgumentOutOfRangeException(...);
   throw;
}
```
Also, I think you will need to delete `_ASSERTE(!IsNilToken(tkType));` from the QCall implementation in C++.

Reply:
Done